lib/chef/provider/mount/linux.rb

```diff
@@ -32,19 +33,29 @@ class Linux < Chef::Provider::Mount::Mount
 
         def mounted?
           mounted = false
-
           real_mount_point = if ::File.exists? @new_resource.mount_point
                                ::File.realpath(@new_resource.mount_point)
                              else
                                @new_resource.mount_point
                              end
 
+          # get the output of losetup command to check loop mount points
+          loop_mount_points = shell_out!("losetup -a").stdout
```

Comment:
This should really only be done when we need it down in the case statement on line 45. Otherwise we're shelling out here for no reason when the first when of the case statement is reached.

Reply:
Hi, as per this comment on earlier PR [11376 comment](https://github.com/chef/chef/pull/11376#discussion_r616854963), its not a good idea to keep the `shell_out` in case-statement, hence i moved it at the top of loop to avoid running statement again and again.

Reply:
How about this approach ?

Reply:
@tas50, please let me know if the above approach looks fine, i will make finalise the changes. Thanks

Reply:
@tas50  Hi, shall I go ahead with this approach ? and update the PR ?